test/cpp/api/autograd.cpp

```diff
@@ -105,6 +106,50 @@ TEST(CustomAutogradTest, MarkNonDifferentiable) {
   y.sum().backward();
 }
 
+TEST(CustomAutogradTest, MarkNonDifferentiableMixed) {
+  struct MyFunction : public Function<MyFunction> {
+    static variable_list forward(AutogradContext *ctx, Variable input) {
+      Variable a = input+1;
+      Variable b = input+2;
+      ctx->mark_non_differentiable({a});
+      return {a,b};
+    }
+
+    static variable_list backward(AutogradContext *ctx, variable_list grad_output) {
+      Variable &grad_a = grad_output[0], &grad_b = grad_output[1];
```

Comment:
Taking a mutable reference here is kind of goofy but I guess it does no harm in this particular block of code.